Data/HashMap/Base.hs

```diff
@@ -286,6 +286,19 @@ instance Eq k => Eq1 (HashMap k) where
     liftEq = equal1
 #endif
 
+-- | Note that, in the presence of hash collisions, equal @HashMap@s may
+-- behave differently, i.e. substitutivity may be violated:
+--
+-- >>> data D = A | B deriving (Eq, Show)
+-- >>> instance Hashable D where hashWithSalt salt d = salt
+--
+-- >>> fromList [(A,1), (B,2)] == fromList [(B,2), (A,1)]
+-- True
+--
+-- >>> fromList [(A,1), (B,2)]
+-- fromList [(A,1),(B,2)]
+-- >>> fromList [(B,2), (A,1)]
+-- fromList [(B,2),(A,1)]
```

Comment:
I don't like using the `Show` instance as an example; the `"fromList"` it adds is confusing. Why not use `toList`?

Reply:
Done. I'm also using variables now, since that seemed to make the examples more readable.